src/main/java/com/google/api/generator/engine/ast/TypeNode.java

```diff
@@ -138,6 +158,24 @@ public int hashCode() {
     return hash;
   }
 
+  private boolean strictEquals(TypeNode other) {
```

Comment:
Consider making this (and `isBoxedTypeEquals`) un-private, and add a `@VisibleForTesting` annotation. That will allow us to add separate tests for clarity.

Reply:
sure, sounds reasonable to me, made the two methods as `protected` and `@VisibleForTesting`. This helps me to tested the two methods separately.